sql/datetimetype.go

```diff
@@ -326,6 +326,22 @@ func (t datetimeType) Equals(otherType Type) bool {
 	return t.baseType == otherType.Type()
 }
 
+// MaxResponseByteLength implements the Type interface
+func (t datetimeType) MaxResponseByteLength() uint32 {
+	switch t.baseType {
+	case sqltypes.Date:
+		return uint32(len(DateLayout))
+	case sqltypes.Time:
+		return uint32(len("+123:00:00.999999"))
+	case sqltypes.Year:
+		return 4
```

Comment:
`sqltypes.Time` and `sqltypes.Year` are handled in their own respective types, so we can remove this here.